r/R/dataset-format.R

```diff
@@ -122,6 +122,31 @@ CsvFileFormat$create <- function(...,
                                  opts = csv_file_format_parse_options(...),
                                  convert_options = csv_file_format_convert_opts(...),
                                  read_options = csv_file_format_read_opts(...)) {
+
+  options <- list(...)
+  schema  <- options[["schema"]]
+
+  column_names     <- read_options$column_names
+  schema_names     <- names(schema)
+
+  if (!is.null(schema) & !identical(schema_names, column_names)) {
+
+    # Element wise comparison and set differnce of column_names and names in schema
+    mismatch_colnames <- column_names[match(column_names, schema_names, nomatch = 0) != seq(column_names)]
+    not_in_schema     <- setdiff(schema_names, column_names)
+
+    abort(c(
+            paste(
+                "column_names not matching or not found in schema-names:",
```

Comment:
What do you think to the idea of rephrasing this to use "should" or "must" as per the tidyverse style guide (https://style.tidyverse.org/error-messages.html)?